File: kitchenStory/src/app/food.ts
import { Category } from "./category";
import { Order } from "./order";

export class Food {

      private  foodId:number;
      private  foodName:string;
      private  foodCategory:string;
       foodPrice:string;
      private category :Category
      quantity:number;
      order:Order;
      constructor(foodName:string,foodCategory:string,foodPrice:string,category:Category)
      {
          this.foodName = foodName;
          this.foodCategory = foodCategory;
          this.foodPrice = foodPrice;
          this.category = category;
      }
      setOrder(order:Order)
      {
         this.order = order;
      }
}

File: kitchen/src/main/java/com/food/kitchen/model/Food.java
package com.food.kitchen.model;


import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.xml.catalog.Catalog;
import java.io.Serializable;

@Entity
@Table(name="Food")
@Data
public class Food  implements Serializable {
    @Id
    @GeneratedValue(strategy = GenerationType.SEQUENCE)
    private Integer foodId;
    private String foodName;
    private String foodCategory;
    private String foodPrice;
    private Integer quantity;
    @OneToOne
    @JoinColumn(name = "category_Id",referencedColumnName = "categoryId")
    private Category category;
    @OneToOne
    @JoinColumn(name = "order_Id",referencedColumnName = "orderId")
    Order order;

}
